12_1_roles.sql

```diff
@@ -1,15 +1,6 @@
-/* create roles */
-DO $$
-DECLARE
-    role text;
-BEGIN
-    FOREACH role IN ARRAY ARRAY['qgep_viewer', 'qgep_user', 'qgep_manager', 'qgep_sysadmin'] LOOP
-      IF NOT EXISTS (SELECT 1 FROM pg_roles WHERE rolname = role) THEN
-          EXECUTE format('CREATE ROLE %1$I NOSUPERUSER INHERIT NOCREATEDB NOCREATEROLE NOREPLICATION', role);
-      END IF;
-    END LOOP;
-END
-$$;
+------------------------------------------
+/* GRANT on schemas - once per database */
+------------------------------------------
 
 /* Viewer */
 GRANT USAGE ON SCHEMA qgep_od  TO qgep_viewer;
```

Comment:
@ponceta are the permissions on the sequences required for a viewer or can that be granted for user only?

Reply:
Everything what is needed to run the calculations for labelling or additional attributes such as _*functionhierarchic has to be granted, else we will have wrong labellings in the view I guess.

Reply:
Do we need sequences for this?

Reply:
To my knowledge it is not required, but I rarely use the qgep_viewer role.